### ISSUE TITLE
Compile Time Error on RHEL 7 on running make all command 

### ISSUE THREAD

**Issue by prakharkohq** [RedisLabsModules/RediSearch#318]
I cloned the project from GIT Repo 
and tried to run make all command on rhel 7
i got this error i am clueless how to get a .so(Shared Object to load into my redis )
Error is this 
cc -Wall -Wno-unused-function -Wno-unused-variable -Wno-unused-result -fPIC -Werror=implicit-function-declaration -D_GNU_SOURCE -std=gnu99 -I"/home/RediSearch/src" -DREDIS_MODULE_TARGET -DREDISMODULE_EXPERIMENTAL_API  -g -ggdb -O3  -Wfree-nonheap-object   -c /home/RediSearch/src/aggregate/aggregate_plan.c -o /home/RediSearch/src/aggregate/aggregate_plan.o -MMD -MF /home/RediSearch/src/aggregate/aggregate_plan.d
cc -Wall -Wno-unused-function -Wno-unused-variable -Wno-unused-result -fPIC -Werror=implicit-function-declaration -D_GNU_SOURCE -std=gnu99 -I"/home/RediSearch/src" -DREDIS_MODULE_TARGET -DREDISMODULE_EXPERIMENTAL_API  -g -ggdb -O3  -Wfree-nonheap-object   -c /home/RediSearch/src/aggregate/aggregate_request.c -o /home/RediSearch/src/aggregate/aggregate_request.o -MMD -MF /home/RediSearch/src/aggregate/aggregate_request.d
In file included from /home/RediSearch/src/aggregate/aggregate_request.c:8:0:
/home/RediSearch/src/aggregate/aggregate_request.c: In function âbuildGroupByâ:
/home/RediSearch/src/util/arr.h:124:7: error: unknown type name â__auto_typeâ
       __auto_type as = arr[i];                      \
       ^
/home/RediSearch/src/aggregate/aggregate_request.c:130:3: note: in expansion of macro âarray_foreachâ
   array_foreach(grp->reducers, red, {
   ^
/home/RediSearch/src/aggregate/aggregate_request.c:130:17: error: incompatible types when initializing type âintâ using type âAggregateGroupReduceâ
   array_foreach(grp->reducers, red, {
                 ^
/home/RediSearch/src/util/arr.h:124:24: note: in definition of macro âarray_foreachâ
       __auto_type as = arr[i];                      \
                        ^
/home/RediSearch/src/aggregate/aggregate_request.c:131:38: error: request for member âreducerâ in something not a structure or union
     Reducer *r = GetReducer(sctx, red.reducer, red.alias, red.args, array_len(red.args), err);
                                      ^
/home/RediSearch/src/util/arr.h:125:7: note: in definition of macro âarray_foreachâ
       blk;                                          \
       ^
/home/RediSearch/src/aggregate/aggregate_request.c:131:51: error: request for member âaliasâ in something not a structure or union
     Reducer *r = GetReducer(sctx, red.reducer, red.alias, red.args, array_len(red.args), err);
                                                   ^
/home/RediSearch/src/util/arr.h:125:7: note: in definition of macro âarray_foreachâ
       blk;                                          \
       ^
/home/RediSearch/src/aggregate/aggregate_request.c:131:62: error: request for member âargsâ in something not a structure or union
     Reducer *r = GetReducer(sctx, red.reducer, red.alias, red.args, array_len(red.args), err);
                                                              ^
/home/RediSearch/src/util/arr.h:125:7: note: in definition of macro âarray_foreachâ
       blk;                                          \
       ^
/home/RediSearch/src/aggregate/aggregate_request.c:131:82: error: request for member âargsâ in something not a structure or union
     Reducer *r = GetReducer(sctx, red.reducer, red.alias, red.args, array_len(red.args), err);
                                                                                  ^
/home/RediSearch/src/util/arr.h:125:7: note: in definition of macro âarray_foreachâ
       blk;                                          \
       ^
/home/RediSearch/src/aggregate/aggregate_request.c:130:32: warning: variable âredâ set but not used [-Wunused-but-set-variable]
   array_foreach(grp->reducers, red, {
                                ^
/home/RediSearch/src/util/arr.h:124:19: note: in definition of macro âarray_foreachâ
       __auto_type as = arr[i];                      \
                   ^
make: *** [/home/RediSearch/src/aggregate/aggregate_request.o] Error 1
[root@cafe-jenkins src]# make all
cc -Wall -Wno-unused-function -Wno-unused-variable -Wno-unused-result -fPIC -Werror=implicit-function-declaration -D_GNU_SOURCE -std=gnu99 -I"/home/RediSearch/src" -DREDIS_MODULE_TARGET -DREDISMODULE_EXPERIMENTAL_API  -g -ggdb -O3  -Wfree-nonheap-object   -c /home/RediSearch/src/aggregate/aggregate_request.c -o /home/RediSearch/src/aggregate/aggregate_request.o -MMD -MF /home/RediSearch/src/aggregate/aggregate_request.d
In file included from /home/RediSearch/src/aggregate/aggregate_request.c:8:0:
/home/RediSearch/src/aggregate/aggregate_request.c: In function âbuildGroupByâ:
/home/RediSearch/src/util/arr.h:124:7: error: unknown type name â__auto_typeâ
       __auto_type as = arr[i];                      \
       ^
/home/RediSearch/src/aggregate/aggregate_request.c:130:3: note: in expansion of macro âarray_foreachâ
   array_foreach(grp->reducers, red, {
   ^
/home/RediSearch/src/aggregate/aggregate_request.c:130:17: error: incompatible types when initializing type âintâ using type âAggregateGroupReduceâ
   array_foreach(grp->reducers, red, {
                 ^
/home/RediSearch/src/util/arr.h:124:24: note: in definition of macro âarray_foreachâ
       __auto_type as = arr[i];                      \
                        ^
/home/RediSearch/src/aggregate/aggregate_request.c:131:38: error: request for member âreducerâ in something not a structure or union
     Reducer *r = GetReducer(sctx, red.reducer, red.alias, red.args, array_len(red.args), err);
                                      ^
/home/RediSearch/src/util/arr.h:125:7: note: in definition of macro âarray_foreachâ
       blk;                                          \
       ^
/home/RediSearch/src/aggregate/aggregate_request.c:131:51: error: request for member âaliasâ in something not a structure or union
     Reducer *r = GetReducer(sctx, red.reducer, red.alias, red.args, array_len(red.args), err);
                                                   ^
/home/RediSearch/src/util/arr.h:125:7: note: in definition of macro âarray_foreachâ
       blk;                                          \
       ^
/home/RediSearch/src/aggregate/aggregate_request.c:131:62: error: request for member âargsâ in something not a structure or union
     Reducer *r = GetReducer(sctx, red.reducer, red.alias, red.args, array_len(red.args), err);
                                                              ^
/home/RediSearch/src/util/arr.h:125:7: note: in definition of macro âarray_foreachâ
       blk;                                          \
       ^
/home/RediSearch/src/aggregate/aggregate_request.c:131:82: error: request for member âargsâ in something not a structure or union
     Reducer *r = GetReducer(sctx, red.reducer, red.alias, red.args, array_len(red.args), err);
                                                                                  ^
/home/RediSearch/src/util/arr.h:125:7: note: in definition of macro âarray_foreachâ
       blk;                                          \
       ^
/home/RediSearch/src/aggregate/aggregate_request.c:130:32: warning: variable âredâ set but not used [-Wunused-but-set-variable]
   array_foreach(grp->reducers, red, {
                                ^
/home/RediSearch/src/util/arr.h:124:19: note: in definition of macro âarray_foreachâ
       __auto_type as = arr[i];                      \
                   ^
make: *** [/home/RediSearch/src/aggregate/aggregate_request.o] Error 1

**Comment by dvirsky** [comment]
yeah we only support gcc 4.9 and up now. you can run it from docker if you want.

**Comment by dvirsky** [comment]
Actually, I've made a change to master that will likely fix this issue. Please pull master and recompile, let me know if it's fixed now.

**Comment by dvirsky** [comment]
@prakharkohq have you tried recompiling it? It should work and I'd like to close this issue. I just don't have that distro here.
